MachineLearningPrototype/utils/preprocess_data.py

```diff
@@ -15,3 +15,24 @@ def extract_quotation_content(content):
     if (len(substring) >= 1):
         result=' '.join(substring)
         return result
+
+def extract_jenkins_link_and_testname(content):
+    pattern = r"((?:http|ftp|https):\/\/.+\/job\/Test_openjdk\d+.+\/\d+)" #For explanation visit https://regexr.com/60jmf
```

Comment:
Currently in our cases, we only query `https` links, so we can ignore `http` and `ftp`. The other parts look good!